cmd/web-handlers.go

```diff
@@ -717,27 +726,28 @@ next:
 				}) {
 					return toJSONError(ctx, errAccessDenied)
 				}
+				// Check if lock mode is set up
+				if bucketObjectLockEnabled {
+					if globalPolicySys.IsAllowed(policy.Args{
+						Action:          policy.BypassGovernanceRetentionAction,
+						BucketName:      args.BucketName,
+						ConditionValues: getConditionValues(r, "", "", nil),
+						IsOwner:         false,
+						ObjectName:      objectName,
+					}) {
+						govBypassPerms = ErrAccessDenied
```

Comment:
policy allowed means the call can go through you have inverted the logic.

Reply:
It is not really inverted, it is restricted to only times, when the bucket/object is protected by an active retention/locking configuration. 

Check is limited to those buckets/objects protected by an active retention/lock configuration. When there is no retention configuration defined, then what is the point of checking and telling user, sorry you cannot delete this object because you are not allowed to override the retention/lock configuration, which is not even defined.

Reply:
No the style of flow is wrong, @ebozduman there is no such inverted logic anywhere in the code.. it looks purposefully wrong.

Reply:
I am working on the inverted logic to bring it back. 
But there is a case when the bucket is not locked, and user has Delete permission, we have to toggle to ErrNone. That I can put in the else part of `if bucketObjectLockEnabled {`

So, if I go back to inverted logic as above, would that be good.
I am making  those changes before creating the new PR